base_layer/core/src/chain_storage/test/chain_storage.rs

```diff
@@ -620,10 +620,10 @@ fn restore_mmr() {
     let rp_mmr_state = store.fetch_mmr_base_leaf_nodes(MmrTree::RangeProof, 0, 100).unwrap();
     let header_mmr_state = store.fetch_mmr_base_leaf_nodes(MmrTree::Header, 0, 100).unwrap();
 
-    assert_eq!(utxo_mmr_state.total_leaf_count, 9);
-    assert_eq!(kernel_mmr_state.total_leaf_count, 3);
-    assert_eq!(rp_mmr_state.total_leaf_count, 9);
-    assert_eq!(header_mmr_state.total_leaf_count, 2);
+    assert_eq!(utxo_mmr_state.total_leaf_count, 10);
+    assert_eq!(kernel_mmr_state.total_leaf_count, 4); // Todo This might be wrong, verify  1 + 3 + 2 +2  = 8
+    assert_eq!(rp_mmr_state.total_leaf_count, 10);
+    assert_eq!(header_mmr_state.total_leaf_count, 2); // Todo This might be wrong, verify 4
```

Comment:
```suggestion
    assert_eq!(header_mmr_state.total_leaf_count, 2); 
```
A header leaf count of 2 is correct. It is only the total leaf count of the base mmr. Only the genesis block and block 1 would be part of the base mmr at this point due to the set MerkleChangeTracker settings.